ec/src/models/twisted_edwards_extended.rs

```diff
@@ -742,15 +740,11 @@ impl<P: Parameters> CanonicalSerialize for GroupAffine<P> {
 
     #[inline]
     fn uncompressed_size(&self) -> usize {
-        Self::UNCOMPRESSED_SIZE
+        // x  + y
+        P::BaseField::zero().serialized_size() + P::BaseField::zero().serialized_size()
```

Comment:
Seems like fields need a trait guarantee that they have constant serialization size, in order for this to be safe?

Reply:
Not sure I understand what you mean?

Reply:
Do you mean that we don't do varint-style encoding?

Reply:
Suppose a field specialized how they serialize `0`. e.g. they serialized `0` with 1 byte `0u8`, and every other element is serialized as `1 + 2 * data_value_in_bits`.

(Or more generally, if a field serialized two elements `a` and `b` in ways such that `serialized_size(a) != serialized_size(b)`, which can arise with varint encodings for instance)

Reply:
Yeah, we need this guarantee (and it's satisfied by all existing field implementations)

Reply:
But I can change it to this to make it more accurate:
```suggestion
        self.x.serialized_size() + self.y.serialized_size()
```